modules/Channel.md

```diff
@@ -0,0 +1,112 @@
+## Channel Module
+
+A channel is a FIFO data structure used to communicate between different Chapel
+tasks. You can learn more about how go-channels work by following this
+[link](https://gobyexample.com/channels)
+Since the operations on the channel are blocking by default, it enables to send
+values from one task to another in a synchronized manner.
+Here is a simple example in Go.
+
+```
+func main() {
+	// make a channel for string values
+	messages := make(chan string)
+
+	// new task to send a value to the channel
+	// it will wait until other task receives
+	go func() { messages <- "ping" }()
+
+	// receive the sent value
+	msg := <-messages
+	fmt.Println(msg)
+}
+```
+
+The module will have a generic `Channel` class (or record) to store the channel
+buffer and support operations on them.
+
+* The main functions that the module should have
+
+	* `init(type eltType, size : int)` : Specify the buffer data type and size
+	for creating a Channel.
+
+	* `send(value : eltType)` : This function waits for a space in the buffer
+	and	sends the `value` to the channel.
+
+	* `recv(out value : eltType) : bool` : This function waits until there is
+	something in the buffer and stores the received entity in `value`. It also
+	returns the status i.e., the operation is executed successfully or not.
+
+	* `close()` : This function is used to close the channel such that no more
+	values will be sent to it.
+
+* `iter these()` : An iterator to receive the values sent to the channel until
+it is closed.
+
+* `select` statements : This statement lets you wait for multiple channel
+operations and executes if anyone of them is ready for communication. `select`
+statements in Go are specifically designed to work with channels.
+Some key points are:
+
+	* `select` statements are commonly used in Go code when working with channels.
+	* The operations are atomic, i.e., checking a channel and performing `recv`
+		or `send` on it is considered as one operation.
+	* The cases are arranged in a random order to prevent starvation.
+
+For e.g.
+```
+// Select syntax in Go to wait on any one of the two channels for receiving
+// If every channel is blocked, the default case is executed.
+select {
+	case msg1 := <-chan1 :
+		fmt.Println("Received", msg1)
+	case msg2 := <-chan2 :
+		fmt.Println("Received", msg2)
+	default :
+		fmt.Println("None")
+}
+```
+
+```
+// Select statements can be used to implement try-send or try-recv
+// default case is executed when none of the cases are ready yet.
+select {
+	case val := <-chan1 :
+		fmt.Println("Received", val);
+	default :
+		fmt.Println("Channel not ready");
+}
+```
+
+### Some important questions related to the module-
+
+* Should this module be standard or package ?
```

Comment:
I tend to think that all modules should start out as packages until they reach a certain level of maturity / number of use cases / reliance that might suggest it should be promoted to a standard module.  That said, the interaction with language syntax through the select statement _might_ make it more appropriate as a standard module than the typical module unless the expanded notion of select statement isn't specific to channels and channels just happen to have the property that it relies upon to get the systems-y definition.